cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct.js

```diff
@@ -362,6 +363,10 @@ function algoliaLocalizedProduct(parameters) {
 
             if (baseModel && baseModel[attributeName]) {
                 this[attributeName] = baseModel[attributeName];
+            } else if (baseModel && parentAttribute && subAttribute && baseModel[parentAttribute] && baseModel[parentAttribute][subAttribute]) {
```

Comment:
Now your latest changes broke this part. `parentAttribute` and `subAttribute` are not defined at this point so the condition is always false.
You really need to write exhaustive test scenarios, ideally manual tests and unit tests.